addons/purchase_mrp/tests/test_anglo_saxon_valuation.py

```diff
@@ -274,6 +275,208 @@ def test_fifo_cost_adjust_mo_quantity(self):
 
         self.assertEqual(self.product_a.standard_price, 100)
 
+    def test_kit_cost_share_variant_bom_validation(self):
+        """
+        Ensure the bom validation work fine with the cost share with optional or variant specific lines
+        """
+        attributes = self.env['product.attribute'].create([
+            {'name': name}
+        for name in ('Size', 'Color')])
+        attributes_values = ((attributes[0], ('S', 'M')), (attributes[1], ('Blue', 'Red')))
+        self.env['product.attribute.value'].create([{
+            'name': name,
+            'attribute_id': attribute.id
+        } for attribute, names in attributes_values for name in names])
+        product_template = self.env['product.template'].create({
+            'name': "tshirt",
+            'is_storable': True,
+        })
+        size_attribute_lines, color_attribute_lines = self.env['product.template.attribute.line'].create([{
+            'product_tmpl_id': product_template.id,
+            'attribute_id': attribute.id,
+            'value_ids': [Command.set(attribute.value_ids.ids)]
+        } for attribute in attributes])
+        self.assertEqual(product_template.product_variant_count, 4)
+
+        c1, c2, c3 = self.env['product.product'].create([
+            {
+                'name': f'Comp {i + 1}',
+                'categ_id': self.avco_category.id,
+            } for i in range(3)
+        ])
+
+        # Two attribute-specific lines whose cost shares both sum to 100% → invalid
```

Comment:
The arrow emoji looks like an AI move. As there is only 110 occurrences of  `→` and 14603 of `->` in the codebase we'll go with the second one.
```suggestion
        # Two complementary attribute-specific lines whose cost shares both sum to 100% -> invalid
```
Same applies for all the other occurrences of the emoji.